lib/UmdMainTemplatePlugin.js

```diff
@@ -2,176 +2,158 @@
-				if(Array.isArray(request)) {
-					expr = "require(" + JSON.stringify(request[0]) + ")" + accessorToObjectAccess(request.slice(1));
-				} else
-					expr = "require(" + JSON.stringify(request) + ")";
-				if(m.optional) {
-					expr = "(function webpackLoadOptionalExternalModule() { try { return " + expr + "; } catch(e) {} }())";
-				}
-				return expr;
-			}).join(", "));
-		}
+			function externalsRequireArray(type) {
+				return replaceKeys(externals.map(m => {
+					let expr;
+					let request = m.request;
+					if(typeof request === "object") request = request[type];
+					if(Array.isArray(request)) {
+						expr = `require(${JSON.stringify(request[0])})${accessorToObjectAccess(request.slice(1))}`;
+					} else
+						expr = `require(${JSON.stringify(request)})`;
+					if(m.optional) {
+						expr = `(function webpackLoadOptionalExternalModule() { try { return ${expr}; } catch(e) {} }())`;
+					}
+					return expr;
+				}).join(", "));
+			}
 
-		function externalsArguments(modules) {
-			return modules.map(function(m) {
-				return "__WEBPACK_EXTERNAL_MODULE_" + m.id + "__";
-			}).join(", ");
-		}
+			function externalsArguments(modules) {
+				return modules.map(m => `__WEBPACK_EXTERNAL_MODULE_${m.id}__`).join(", ");
+			}
 
-		function libraryName(library) {
-			return JSON.stringify(replaceKeys([].concat(library).pop()));
-		}
+			function libraryName(library) {
+				return JSON.stringify(replaceKeys([].concat(library).pop()));
+			}
 
-		var amdFactory;
-		if(optionalExternals.length > 0) {
-			amdFactory = "function webpackLoadOptionalExternalModuleAmd(" + externalsArguments(requiredExternals) + ") {\n" +
-				"			return factory(" + (
-					requiredExternals.length > 0 ?
-					externalsArguments(requiredExternals) + ", " + externalsRootArray(optionalExternals) :
-					externalsRootArray(optionalExternals)
-				) + ");\n" +
-				"		}";
-		} else {
-			amdFactory = "factory";
-		}
+			let amdFactory;
+			if(optionalExternals.length > 0) {
+				amdFactory = `function webpackLoadOptionalExternalModuleAmd(${externalsArguments(requiredExternals)}) {\n			return factory(${requiredExternals.length > 0 ?
+externalsArguments(requiredExternals) + ", " + externalsRootArray(optionalExternals) :
+externalsRootArray(optionalExternals)});\n		}`;
+			} else {
+				amdFactory = "factory";
+			}
 
-		return new ConcatSource(new OriginalSource(
-			"(function webpackUniversalModuleDefinition(root, factory) {\n" +
+			return new ConcatSource(new OriginalSource(
+				`(function webpackUniversalModuleDefinition(root, factory) {\n${this.auxiliaryComment &&
+			typeof this.auxiliaryComment === "string" ?
+			"   //" + this.auxiliaryComment + "\n" :
+			this.auxiliaryComment.commonjs2 ?
+			"   //" + this.auxiliaryComment.commonjs2 + "\n" :
+			""}    if(typeof exports === 'object' && typeof module === 'object')\n		module.exports = factory(${externalsRequireArray("commonjs2")});\n${this.auxiliaryComment &&
```

Comment:
The spacing after `""}` must be a Tab instead of spaces.

The formating of this part looks a bit weird. I was confused by the `""}` on line start. It would be great to have a line break before each line in the generate code. (after `\n`)